src/Features/Core/Portable/Diagnostics/DiagnosticService.cs

```diff
@@ -40,6 +46,9 @@ public DiagnosticService(IAsynchronousOperationListenerProvider listenerProvider
 
             _gate = new object();
             _map = new Dictionary<IDiagnosticUpdateSource, Dictionary<Workspace, Dictionary<object, Data>>>();
+
+            _eventListenerTracker = new EventListenerTracker<IDiagnosticService>(
+                eventListeners.Where(l => l.Metadata.Service == WellKnownEventListeners.DiagnosticService));
```

Comment:
Can the constructor just take the string kind and do this filtering? That way you don't have to do it in multiple places.

Reply:
sure.